client/lib/query-manager/post/index.js

```diff
@@ -0,0 +1,155 @@
+/**
+ * External dependencies
+ */
+import moment from 'moment';
+import every from 'lodash/every';
+import some from 'lodash/some';
+import includes from 'lodash/includes';
+import get from 'lodash/get';
+
+/**
+ * Internal dependencies
+ */
+import QueryManager from '../';
+import PostQueryKey from './key';
+import { DEFAULT_POST_QUERY } from './constants';
+
+/**
+ * PostQueryManager manages posts which can be queried and change over time
+ */
+export default class PostQueryManager extends QueryManager {
+	/**
+	 * Returns true if the post matches the given query, or false otherwise.
+	 *
+	 * @param  {Object}  query Query object
+	 * @param  {Object}  post  Item to consider
+	 * @return {Boolean}       Whether post matches query
+	 */
+	matches( query, post ) {
+		const queryWithDefaults = Object.assign( {}, DEFAULT_POST_QUERY, query );
+		return every( queryWithDefaults, ( value, key ) => {
+			switch ( key ) {
+				case 'search': {
+					const search = new RegExp( value, 'i' );
+					return search.test( post.title ) || search.test( post.content );
+				}
+
+				case 'after':
+				case 'before':
+				case 'modified_after':
+				case 'modified_before': {
+					const queryDate = moment( value, moment.ISO_8601 );
+					const comparison = /after$/.test( key ) ? 'isAfter' : 'isBefore';
+					const field = /^modified_/.test( key ) ? 'modified' : 'date';
+					return queryDate.isValid() && moment( post[ field ] )[ comparison ]( queryDate );
+				}
+
+				case 'tag':
+				case 'category': {
+					const search = new RegExp( `^${ value }$`, 'i' );
```

Comment:
>Are we supporting regex search, or should this be escaped?
 
No, we shouldn't support regex, only case-insensitive. Changes `search`, `tag`, and `category` to match on lowercase comparison in rebased 7c21ef133bb74fb2f8b684dbb75b6a3ec1b5442e ([line](https://github.com/Automattic/wp-calypso/pull/5135/commits/7c21ef133bb74fb2f8b684dbb75b6a3ec1b5442e#diff-4fe99528e32f35248cd67da8063b2ca5R32))